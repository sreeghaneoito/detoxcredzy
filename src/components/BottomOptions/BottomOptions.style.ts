import {Dimensions, StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    height: 84,
    width: Dimensions.get('window').width,
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingHorizontal: 24,
    alignItems: 'center',
  },
  customStyle: {
    width: '45%',
  },
  fullWidthBtn: {
    width: '100%',
  },
  shadowIOS: {
    shadowColor: '#00000014',
    shadowOffset: {width: -4, height: 4},
    shadowOpacity: 1,
    shadowRadius: 24,
  },
  shadowAndroid: {
    elevation: 36,
    shadowColor: 'black',
  },
  // marginIos: {
  //   marginBottom: 84,
  // },
  // marginAndroid: {
  //   marginBottom: 50,
  // },
});
