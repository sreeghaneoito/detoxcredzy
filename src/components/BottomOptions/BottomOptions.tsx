import {Platform, View} from 'react-native';
import React from 'react';
import {style} from './BottomOptions.style';
import Button from '../Button/Button.component';
import {useUtils} from '../../hooks/useUtils';
import {BottomOptionProps} from '../../interfaces/component.interface';

/**
 * Bottom Options
 * @param buttonAction : Function trigger on primary button click
 * @param ghostAction: Function trigger on ghost button click
 * @param mainlabel: The Primary Button label
 * @param ghostlabel: The ghost Button label
 * @param loading: Is the Primary button loading or not
 * @param backButtonDisabled: Is the bacButton Disabled or not
 * @returns JSX component
 */
const BottomOptions = ({
  buttonAction,
  ghostAction,
  mainlabel,
  ghostlabel,
  loading,
  backButtonDisabled,
}: BottomOptionProps) => {
  const {theme} = useUtils();

  try {
    return (
      <>
        <View
          style={[
            style.container,
            theme.secondaryExtend,
            Platform.OS === 'ios' ? [style.shadowIOS] : [style.shadowAndroid],
          ]}>
          <Button
            label={ghostlabel}
            type="ghost"
            Customstyle={style.customStyle}
            triggerclick={ghostAction}
            testid="back"
            disabled={backButtonDisabled}
          />
          <Button
            label={mainlabel}
            type="primary"
            Customstyle={style.customStyle}
            triggerclick={buttonAction}
            testid="continue"
            loading={loading}
          />
        </View>
      </>
    );
  } catch (error) {
    console.log('Bottom option crashed =>', error);
    return <View style={style.container} />;
  }
};

export default BottomOptions;
