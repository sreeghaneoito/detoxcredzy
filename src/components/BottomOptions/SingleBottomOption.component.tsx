import {Platform, View} from 'react-native';
import React from 'react';
import {style} from './BottomOptions.style';
import Button from '../Button/Button.component';
import {useUtils} from '../../hooks/useUtils';
import {BottomOptionProps} from '../../interfaces/component.interface';

/**
 * Bottom Options
 * @param buttonAction : Function trigger on primary button click
 * @param mainlabel: The Primary Button label
 * @param loading: Is the Primary button loading or not
 * @returns JSX component
 */
const SingleBottomOption = ({
  buttonAction,
  mainlabel,
  loading,
}: BottomOptionProps) => {
  const {theme} = useUtils();

  try {
    return (
      <>
        <View
          style={[
            style.container,
            theme.secondaryExtend,
            Platform.OS === 'ios' ? [style.shadowIOS] : [style.shadowAndroid],
          ]}>
          <Button
            label={mainlabel}
            type="primary"
            Customstyle={style.fullWidthBtn}
            triggerclick={buttonAction}
            testid="continue"
            loading={loading}
          />
        </View>
      </>
    );
  } catch (error) {
    console.log('Bottom option crashed =>', error);
    return <View style={style.container} />;
  }
};

export default SingleBottomOption;
