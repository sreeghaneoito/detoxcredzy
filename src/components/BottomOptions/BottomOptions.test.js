import React from 'react';
import renderer from 'react-test-renderer';
import BottomOptions from './BottomOptions';

test('renders correctly', () => {
  const bottomoptions = renderer.create(<BottomOptions />).toJSON();
  expect(bottomoptions).toMatchSnapshot();
});
