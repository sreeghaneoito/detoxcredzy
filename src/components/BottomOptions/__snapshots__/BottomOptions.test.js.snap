// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flexDirection": "row",
        "height": 84,
        "justifyContent": "space-around",
        "paddingHorizontal": 24,
        "position": "absolute",
        "width": 750,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Array [
        Object {
          "shadowColor": "#00000014",
          "shadowOffset": Object {
            "height": 4,
            "width": -4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 24,
        },
      ],
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F8F8FA",
        "borderRadius": 4,
        "height": 48,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 16,
        "width": "45%",
      }
    }
    testID="backbutton"
  >
    <Text
      collapsable={false}
      style={
        Object {
          "color": "#000000",
          "fontSize": 13,
          "fontWeight": "500",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#28AF51",
        "borderRadius": 4,
        "height": 48,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 16,
        "width": "45%",
      }
    }
    testID="continuebutton"
  >
    <Text
      collapsable={false}
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 13,
          "fontWeight": "500",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
</View>
`;
